syntax = "proto3";
package related_resource;





/*
UpdateRelatedResource请求
*/
message UpdateRelatedResourceRequest {
message RelatedInfo {
    /*
    id
    */
    string id = 1 ;
    /*
    描述
    */
    string label = 2 ;
    /*
    query条件key
    */
    string reverseQueryKey = 3 ;
}
        
    
    /*
    实例ID
    */
    string instanceId = 1 ;
    /*
    关联名称
    */
    string name = 2 ;
    /*
    关联模型Id
    */
    string relatedObjectId = 3 ;
    /*
    query参数,预留字段
    */
    string query = 4 ;
    /*
    资源模型路径实例查询key信息
    */
    RelatedInfo relatedInfo = 5 ;
}

/*
UpdateRelatedResource返回
*/
message UpdateRelatedResourceResponse {
    /*
    实例ID
    */
    string instanceId = 1 ;
}

/*
UpdateRelatedResourceApi返回
*/
message UpdateRelatedResourceResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    related_resource.UpdateRelatedResourceResponse data = 4 ;
}

